### ISSUE TITLE
CNN custom implementation 

### ISSUE THREAD

**Issue by joaquinCaceres** [speechbrain/speechbrain#960]
Hello, 

What are the differences between CNN custom implementation (Conv1d, BatchNorm1d...) and PyTorch implementation (torch.nn.Conv1d, torch.nn.BatchNorm1d...).  I've noticed that custom cnn are used in the ECAPA-TDNN model.  Is the output of custom layers the same as PyTorch?

Thanks!

**Comment by TParcollet** [comment]
Hi, it is mostly linked to: 1. input shape might differ; 2. Padding is handled in SpeechBrain. But internally, we call the same torch.conv layers. It's just a matter of compliance with the input Tensor format of SpeechBrain that is the same across ALL the layers (not the case for Torch ..)

**Comment by joaquinCaceres** [comment]
Thanks, @TParcollet!
